lib/internal/test_runner/mock/mock.js

```diff
@@ -62,8 +64,63 @@ class MockFunctionContext {
     validateFunction(implementation, 'implementation');
     const nextCall = this.#calls.length;
     const call = onCall ?? nextCall;
-    validateInteger(call, 'onCall', nextCall);
-    this.#mocks.set(call, implementation);
+    // eslint-disable-next-line eqeqeq
+    if (onCall != undefined) {
+      validateInteger(call, 'onCall', nextCall);
+    }
+
+    // eslint-disable-next-line eqeqeq
+    if (onCall == undefined) {
+      this.#addMockToFirstEmptyImpl(implementation);
+      return this;
+    }
+
+    const lastCall = this.#queue[this.#queue.length - 1]?.index;
```

Comment:
same (ArrayPrototypeAt)
```suggestion
    const lastCall = ArrayPrototypeAt(this.#queue, -1)?.index;
```